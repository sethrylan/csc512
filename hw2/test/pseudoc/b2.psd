// source: http://www.ceng.metu.edu.tr/courses/ceng444/link/f3jasmintutorial.html
VAR x:INT;
BEGIN
	READ(x);
	x:=x+3;
	WRITE(x);
END
