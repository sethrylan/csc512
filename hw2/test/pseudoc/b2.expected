.source b2.psd
.class public b2
.super java/lang/Object
;
; standard initializer (calls java.lang.Object's initializer)
;
.method public <init>()V
	aload_0
	invokespecial java/lang/Object/<init>()V
	return
.end method
;
; main()
;
.method public static main([Ljava/lang/String;)V
	.limit stack 5
	.limit locals 100
	ldc 0
	istore 1
	ldc 0
	istore 50
 Label1:
	getstatic java/lang/System/in Ljava/io/InputStream;
	invokevirtual java/io/InputStream/read()I
	istore 51
	iload 51
	ldc 10
	isub
	ifeq Label2
	iload 51
	ldc 32
	isub
	ifeq Label2
	iload 51
	ldc 48
	isub
	ldc 10
	iload 50
	imul
	iadd
	istore 50 
	goto Label1 
  Label2:
	iload 50
	istore 1
	iload 1 
	ldc 3 
	iadd 
	istore 1       ; x=x+3 
	iload 1 
	getstatic java/lang/System/out Ljava/io/PrintStream; 
	swap 
	invokevirtual java/io/PrintStream/println(I)V	; print x
	; done
	return
.end method
